docs/scripts/gen_additional_readmes.py

```diff
@@ -0,0 +1,87 @@
+"""
+This script copies README files into the PrairieLearn documentation.
+It processes the README files from specified source directories, modifies links to point
+to the corresponding files on GitHub, and creates a navigation structure for the documentation.
+"""
+
+import re
+from pathlib import Path
+from types import SimpleNamespace
+
+import mkdocs_gen_files
+
+ROOT = Path.cwd()
+SOURCE_ROOT = "https://github.com/PrairieLearn/PrairieLearn/blob/master/"
```

Comment:
If an image is added to the repo and one of these Markdown files in a PR, this will leave us with an invalid preview as they won't exist on `master` yet.

Reply:
For images, yeah, I think copying makes sense. I'm not 100% sure what you mean by "out-of-tree files/directories", can you give some examples?

Reply:
Let's say I link to a file in a separate directory, e.g. a `.ts` file. This file may not exist on master yet, so the link would be broken locally.

Reply:
What happens if such a link is included completely independently of this PR? If this is a more general problem, I don't really care about what happens in this PR as long as we're consistent.

Reply:
`mkdocs` doesn't let you link to files outside the `docs` directory typically, so I guess this is a non-issue.

Reply:
In 95f5970, I add the image copying logic.